gpu_reliability/platforms/gcp.py

```diff
@@ -105,8 +107,10 @@ def launch_instance(self, request: LaunchRequest):
         # Wait for the create operation to complete.
         self.logger.info(f"Creating instance `{instance_name}`...")
 
-        operation = self.instance_client.insert(request=create_request)
         start = time()
+        operation = self.instance_client.insert(request=create_request)
+        wait_func = functools.partial(compute_v1.ZoneOperationsClient().wait, operation=operation.name, zone=request.geography, project=self.project_id)
+        operation._refresh = wait_func
```

Comment:
Took me a bit of digging to find the caller of this refresh signature, since it looks undocumented in the API lib. Mind adding a comment linking to:

https://github.com/googleapis/python-api-core/blob/9abc6f48f23c87b9771dca3c96b4f6af39620a50/google/api_core/extended_operation.py#L142

Just to track the implementation until upstream is patched?

Reply:
It took me a bit of digging to figure out how to wedge this into the client library too. I've written a comment and pointed at the link you suggested.